File: 프로그래머스/unrated/181918. 배열 만들기 4/배열 만들기 4.java
import java.util.*;
class Solution {
    public Stack<Integer> solution(int[] arr) {
        Stack<Integer> ret = new Stack<>();
        int i = 0; 
        while(i < arr.length){
            if(ret.isEmpty()){
                ret.push(arr[i]);
                i++;
            } else {
                if(ret.peek() < arr[i]){
                    ret.push(arr[i]);
                    i++;
                } else {
                    ret.pop();
                }
            }
        }
        return ret;
    }
}

File: 프로그래머스/0/181918. 배열 만들기 4/배열 만들기 4.py
def solution(arr):
    i = 0
    stk = []
    while i < len(arr):
        if not stk:
            stk.append(arr[i])
            i += 1
        else:
            if stk[-1] < arr[i]:
                stk.append(arr[i])
                i += 1
            else:
                if stk: stk.pop()
    return stk